packages/block-library/src/post-date/edit.js

```diff
@@ -2,25 +2,93 @@
  * WordPress dependencies
  */
 import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import { useState } from '@wordpress/element';
 import { __experimentalGetSettings, dateI18n } from '@wordpress/date';
+import { BlockControls, InspectorControls } from '@wordpress/block-editor';
+import {
+	ToolbarGroup,
+	ToolbarButton,
+	Popover,
+	DateTimePicker,
+	PanelBody,
+	CustomSelectControl,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
-function PostDateDisplay() {
-	const [ date ] = useEntityProp( 'postType', 'post', 'date' );
+function PostDateEditor( { format, setAttributes } ) {
+	const [ siteFormat ] = useEntityProp( 'root', 'site', 'date_format' );
+	const [ date, setDate ] = useEntityProp( 'postType', 'post', 'date' );
+	const [ isPickerOpen, setIsPickerOpen ] = useState( false );
 	const settings = __experimentalGetSettings();
-
+	const is12Hour = /a(?!\\)/i.test(
+		settings.formats.time
+			.toLowerCase()
+			.replace( /\\\\/g, '' )
+			.split( '' )
+			.reverse()
+			.join( '' )
+	);
+	const formatOptions = Object.keys( settings.formats ).map(
+		( formatKey ) => ( {
+			key: settings.formats[ formatKey ],
+			name: dateI18n( settings.formats[ formatKey ], date ),
+		} )
```

Comment:
Might be a good fit for `_.map()` since we're only using `settings.formats`' values? (Or `Object.values().map()`, in case we're polyfilling for older browsers?)

Reply:
Sure, that makes sense.